embeddings/model/lightning_module/question_answering.py

```diff
@@ -0,0 +1,161 @@
+
+from embeddings.data.datamodule import HuggingFaceDataset
+from embeddings.data.dataset import Dataset
+from embeddings.data.io import T_path
+from embeddings.model.lightning_module.huggingface_module import HuggingFaceLightningModule
+from embeddings.model.lightning_module.lightning_module import LightningModule
+
+QA_PREDICTED_ANSWER_TYPE = Dict[str, Union[str, int, float]]
+QA_GOLD_ANSWER_TYPE = Dict[str, Union[Dict[str, Union[List[str], List[Any]]], str, int]]
+
+
+class QuestionAnsweringInferenceModule(pl.LightningModule):
+    def __init__(self, model_name: str) -> None:
+        super().__init__()
+        self.model = AutoModelForQuestionAnswering.from_pretrained(model_name)
+        self.trainer = pl.Trainer(devices="auto", accelerator="auto")
+
+    def predict_step(
+            self, batch: Dict[str, torch.Tensor], batch_idx: int, dataloader_idx: Optional[int] = None
+    ) -> Dict[str, Union[QuestionAnsweringModelOutput, Dict[str, torch.Tensor]]]:
+        outputs = self.model(**batch)
+        return {"data": batch, "outputs": outputs}
+
+    def predict(
+            self,
+            dataloaders: Union[
+                DataLoader[HuggingFaceDataset], Sequence[DataLoader[HuggingFaceDataset]]
+            ],
+    ) -> List[Dict[str, Union[QuestionAnsweringModelOutput, Dict[str, torch.Tensor]]]]:
+        assert self.trainer is not None
+        return self.trainer.predict(  # type: ignore
+            model=self,
+            dataloaders=dataloaders,
+            return_predictions=True,
+        )
+
+
+class QuestionAnsweringModule(LightningModule[AutoModelForQuestionAnswering]):
+    downstream_model_type = AutoModelForQuestionAnswering
+
+    def __init__(
+            self,
+            model_name_or_path: T_path,
+            finetune_last_n_layers: int,
+            config_kwargs: Optional[Dict[str, Any]] = None,
+            task_model_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> None:
+
+        super().__init__(metrics=None, **task_model_kwargs if task_model_kwargs else {})
+        self.save_hyperparameters({"downstream_model_type": self.downstream_model_type.__name__})
+        self.downstream_model_type = self.downstream_model_type
+        self.config_kwargs = config_kwargs if config_kwargs else {}
+        self.target_names: Optional[List[str]] = None
+        self._init_model()
+
+    def _init_metrics(self) -> None:
+        pass
+
+    def _init_model(self) -> None:
+        self.config = AutoConfig.from_pretrained(
+            self.hparams.model_name_or_path,
+            **self.config_kwargs,
+        )
+        self.model: AutoModel = self.downstream_model_type.from_pretrained(
+            self.hparams.model_name_or_path, config=self.config
+        )
+        if self.hparams.finetune_last_n_layers > -1:
+            self.freeze_transformer(finetune_last_n_layers=self.hparams.finetune_last_n_layers)
+
+    def freeze_transformer(self, finetune_last_n_layers: int) -> None:
```

Comment:
To refactor. Code duplication with `HuggingFaceLightningModule.` However, refactoring won't be so fast because it needs changes in the `HuggingFaceLightningModule` logic.

Reply:
@ktagowski @ktagowski, if you decide not to correct it not, we should create a follow-up issue for it. I would go with the follow-up issue for now. Leaving decision to you guys.

Reply:
I agree we should add follow-up issue and handle it seperately